[ernestlsy] iP

ernestlsy opened the pull request:
# Pixel
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Pixel frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/ernestlsy/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

<hr>

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Pixel {
    public static void main(String[] args) {
        new Pixel().run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ernestlsy committed:
Renamed Duke to Pixel. Added greeting and exit message.

ernestlsy committed:
Added echo functionatlity.

ernestlsy committed:
Enabled adding tasks and displaying tasks in list.

ernestlsy committed:
Enabled marking and unmarking of tasks.

ernestlsy committed:
Added support for tracking 3 types of tasks (ToDo, Deadline, Event)

ernestlsy committed:
Added automated text UI testing, touched up text formatting.

ernestlsy committed:
Added handling of input errors.

ernestlsy committed:
Enabled deleting of tasks from list.

ernestlsy committed:
Enable automatically saving task list contents

Contents of task list are lost after program terminates

It would be better to have records of the task list history after termination

Let's write the contents of the task list to a file
(path: "./data/Pixel.txt") after every update to the task list

That way, the task list contents will be saved and can be retrieved after program termination

ernestlsy committed:
Merge branch 'branch-Level-7'

ernestlsy committed:
Event / Deadline classes: Add support for understanding dates and times

Dates and times are treated and recorded as strings

This prevents us from working with dates and times in a meaningful way,
such as calculating remaining time or duration of event

Let's add parsing of input date/time into a LocalDateTime object
Input must now follow the format "YYYY-MM-DD HH:mm"
HH:mm is optional, and defaults to 00:00 if not provided
DateTime is printed in the format "MMM d yyyy HH:mm"

ernestlsy committed:
Merge branch 'branch-Level-8'

namitdeb739 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
```
Remember to add header comments!

namitdeb739 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+        if (comp.length > 2) {
+            throw new PixelException("Please use the format YYYY-MM-DD HH:MM for the date and time!");
+        }
+        try {
```
I like this use of exception handling

namitdeb739 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+    public Deadline(String desc, LocalDateTime dueBy) {
+        super(desc);
+        this.dueBy = dueBy;
+    }
+    public Deadline(String[] comp) throws PixelException {
```
Remember to add line breaks to improve readability

namitdeb739 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,76 @@
+public class Event extends Task {
+    LocalDateTime from;
+    LocalDateTime to;
+    private static LocalDateTime parseDateTime(String input) throws PixelException {
```
Considering the repeated code with this and `Deadline` perhaps you could consider moving this method to a parent and using inheritance?

namitdeb739 commented on src/main/java/Pixel.java:
```diff
@@ -0,0 +1,63 @@
+                        continue;
+                    default:
+                }
+                switch (input) {
+                    case "bye":
+                        wrapPrint(bye);
+                        sc.close();
+                        return;
+                    case "list":
+                        wrapPrint(tasklist.toString());
+                        break;
+                    default:
+                        throw PixelException.unknownInput();
```
Remember, switch cases should have the same indentation level as the switch statement

namitdeb739 reviewed (commented):
LGTM. Just a few nits to fix.

ernestlsy committed:
Add new classes to categorize functionalities

ernestlsy committed:
Organize classes into packages labelled for their general functionality

ernestlsy committed:
Merge branch 'branch-Gradle'

ernestlsy committed:
Edit build.gradle to work for Pixel instead of Duke

ernestlsy committed:
Add unit testing for Parser and TaskList classes

ernestlsy committed:
Add JavaDoc comments to non-private methods

ernestlsy committed:
Reformat code to comply with given coding standard

ernestlsy committed:
Add support for finding tasks matching with keyword

ernestlsy committed:
Merge branch 'branch-A-JavaDoc'

ernestlsy committed:
Merge branch 'branch-A-CodingStandard'

ernestlsy committed:
Merge branch 'branch-Level-9'

ceilingFan456 commented on src/main/java/pixel/command/Command.java:
```diff
@@ -0,0 +1,24 @@
+ * Represents the structure of a Command entered by the user, which performs a function when executed.
+ */
+abstract public class Command {
+    protected boolean isExitCommand = false;
```
really good naming here! i saw some other repo named as continue/status etc. but this name is much clearer!

ceilingFan456 commented on src/test/java/pixel/task/TaskListTest.java:
```diff
@@ -0,0 +1,57 @@
+public class TaskListTest {
+
+    @Test
+    public void testAddTask() {
```
Hi, i think the naming here could be clearer. The textbook suggests `unitBeingTested_descriptionOfTestInputs_expectedOutcome()`. thus maybe it can be renamed as `addTask_oneTask_success()`

ceilingFan456 commented on src/test/java/pixel/util/ParserTest.java:
```diff
@@ -0,0 +1,64 @@
+public class ParserTest {
+
+    @Test
+    public void testParseFullCommand() {
```
Similarly for this one, I think the naming could be rephrased to `parseFullCommand_invalidTaskNumber_exceptionThrown()`.

ceilingFan456 reviewed (commented):
Hi i think all namings are really clear and consistent with textbook specifications. There is no obvious other coding standard violation also. Really good job with the clear code :D

ernestlsy committed:
Add support for checkstyle

Reformat code to comply with given code style

ernestlsy committed:
Merge branch 'branch-A-CheckStyle'

ernestlsy committed:
Implement GUI for Pixel using JavaFX

ernestlsy committed:
Merge branch 'branch-Level-10' to implement GUI

ernestlsy committed:
Refactor code to match implementation of JavaFX

Remove legacy methods and update JavaDoc comments
to reflect changes in functionality of new methods

ernestlsy committed:
Add assertions to verify correctness of TaskList methods

ernestlsy committed:
Improve code readability

Some of the condition checks are overly complicated and
some methods in Parser class are too long

This unnecessary complexity makes the code harder to read

Let's refactor the code by abstracting out some of the condition checks
and change how the long methods are implemented to avoid duplicating code

This will make the code more readable for future maintainers

ernestlsy committed:
Merge pull request #1 from ernestlsy/branch-A-Assertions

Add assertions to verify correctness of TaskList methods

ernestlsy committed:
Sync with master branch

ernestlsy committed:
Merge branch 'master' into branch-A-CodeQuality

ernestlsy committed:
Merge pull request #3 from ernestlsy/branch-A-CodeQuality

Improve code readability

ernestlsy committed:
Enable operations on multiple tasks at once

Tasks can only be marked, unmarked, or deleted one at a time

This is inefficient if the user wants to perform the same operation on multiple tasks

Let's allow the user to mark or unmark multiple consecutive tasks,
and delete all completed tasks, with a single command

This will make user operations more efficient

New commands:
- markAll <from> <to> - Mark all tasks from <from> to <to> inclusive as done
- unmarkAll <from> <to> - Unmark all tasks from <from> to <to> inclusive as not done
- clear - Delete all completed tasks

ernestlsy committed:
Merge branch 'branch-BCD-Extension'

ernestlsy committed:
Remove case-sensitivity for command keywords

ernestlsy committed:
Improve GUI clarity and aesthetics

ernestlsy committed:
Add User Guide
